src/couch_replicator_scheduler.erl

```diff
@@ -205,32 +215,36 @@ handle_config_terminate(Self, _, _) ->
 
 %% private functions
 
-start_jobs(Count) ->
+start_jobs(State, Count) ->
     Runnable0 = pending_jobs(),
     Runnable1 = lists:sort(fun oldest_job_first/2, Runnable0),
     Runnable2 = lists:filter(fun not_recently_crashed/1, Runnable1),
     Runnable3 = lists:sublist(Runnable2, Count),
-    lists:foreach(fun start_job_int/1, Runnable3).
+    [start_job_int(State, Job) || Job <- Runnable3].
 
 
-stop_jobs(Count) ->
+stop_jobs(State, Count) ->
     Running0 = running_jobs(),
     Running1 = lists:sort(fun oldest_job_first/2, Running0),
     Running2 = lists:sublist(Running1, Count),
-    lists:foreach(fun stop_job_int/1, Running2).
+    [stop_job_int(State, Job) || Job <- Running2].
 
 
 oldest_job_first(#job{} = A, #job{} = B) ->
     last_started(A) =< last_started(B).
 
 
 not_recently_crashed(#job{} = Job) ->
-    case crash_history(Job) of
+    case Job#job.history of
         [] ->
             true;
+        [{stopped, _When} | _] ->
+            true;
         [{crashed, When} | _] ->
-            timer:now_diff(os:timestamp(), When)
-                >= ?MINIMUM_CRASH_INTERVAL
+            Crashes = crash_history(Job),
+            BackoffExp = erlang:min(length(Crashes) - 1, ?MAX_BACKOFF_EXPONENT),
+            BackoffInterval = (1 bsl BackoffExp) * ?BACKOFF_INTERVAL_MICROS,
+            timer:now_diff(os:timestamp(), When) >= BackoffInterval
```

Comment:
`{started, _When}` the other pr did `EventsBeforeStop = lists:takewhile` to jump over starts. 

We use starts in `last_started/`1 function. And use it to sort jobs by start order. Wonder if `last_started` should be job record parameter? So we don't have to jump over it in the history. And history is just crash/stop history.